src/reconcile/rec_write.c

```diff
@@ -1395,8 +1395,10 @@ __wt_rec_split_finish(WT_SESSION_IMPL *session, WT_RECONCILE *r)
      *
      * Pages with skipped or not-yet-globally visible updates aren't really empty; otherwise, the
      * page is truly empty and we will merge it into its parent during the parent's reconciliation.
+     *
+     * If checkpoint has no entry left to write, reconciliation is done.
      */
-    if (r->entries == 0 && r->supd_next == 0)
+    if (r->entries == 0 && (r->supd_next == 0 || F_ISSET(r, WT_REC_CHECKPOINT)))
```

Comment:
Before durable history, the code is not checking for `checkpoint` and it is passing. With durable history changes in function `__rec_split_write` may be the reason behind the problem. Even if the check is correct, why the issue doesn't occur in the eviction flow? As page may not get evicted. But, when the parent page gets reconciled the same issue, can happen right?

Reply:
For eviction, we can't ignore the empty page if there are uncommitted changes to it. The leaf page and the parent page will be kept in memory by update restore. So the problem doesn't happen.